source/retryable-writes/retryable-writes.rst

```diff
@@ -66,9 +66,23 @@ ClientSession
    name of this object MAY vary across drivers.
 
 Retryable Error
-   An error is considered retryable if it meets any of the following criteria:
+   An error is considered retryable if it has a RetryableWriteError label in
+   its top-level "errorLabels" field.
 
-   - any network exception (e.g. socket timeout or error)
+   For server versions 4.4 and newer, MongoDB will add a RetryableWriteError label to
+   errors or server responses that it considers retryable before returning them to the
+   driver. As new server versions are released, the errors that are labeled with the
+   RetryableWriteError may change. When receiving a command result with an error from
+   a 4.4+ server that supports retryable writes, the driver MUST NOT add a
+   RetryableWriteError label to that error under any condition.
+
+   When the driver encounters a network error communicating with any server version
+   that supports retryable writes, it MUST add a RetryableWriteError label to that
+   error.
+
+   When receiving a command result with an error from a pre-4.4 server that supports
+   retryable writes, the driver MUST add a RetryableWriteError label to errors that meet
+   the following criteria:
```

Comment:
@jmikola Let me know what you think of this section now. I removed all mentions of the "not master" and "node is recovering" messages because people mentioned in the meeting that those had been eliminated before 3.6, but let me know if you think I should add them back in the pre-4.4 section?

Reply:
I'm OK with this change, but I think other reviewers should sign off on it as well.

I might also suggest either calling this out in a new [Q&A entry](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#q-a) to explain why we're removing this. For example:

> Why did an earlier version of the spec parse errmsg to categorize retryable errors?

The answer could mention that the spec originally copied that behavior from SDAM. You can reference the [Use error messages to detect "not master" and "node is recovering"](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#use-error-messages-to-detect-not-master-and-node-is-recovering) section specifically and clarify that retryable writes is 3.6+ and we can safely rely on those servers to use consistent error codes. I believe that was the take-away from our meeting last week with server folks.

Reply:
Okay, happy to do that.